modules/oke/nodepools.tf

```diff
@@ -11,6 +11,7 @@ resource "oci_containerengine_node_pool" "nodepools" {
   name               = var.label_prefix == "none" ? each.key : "${var.label_prefix}-${each.key}"
 
   freeform_tags = var.freeform_tags["node_pool"]
+  defined_tags  = merge(var.defined_tags["node_pool"], lookup(each.value, "defined_tags", {}))
```

Comment:
Am I right in understanding that you'll be creating a `defined_tags` for each node_pool but with potentially a default one for all of them? Something like this:

```yaml
np1 = { shape = "VM.Standard.E4.Flex", ocpus = 2, memory = 32, node_pool_size = 2, boot_volume_size = 150, label = { app = "frontend", pool = "np1" }, defined_tags = { app = "frontend", pool = "np1" } }

np2 = { shape = "VM.Standard.E4.Flex", ocpus = 2, memory = 32, node_pool_size = 2, boot_volume_size = 150, label = { app = "frontend", pool = "np2" } }
```

If yes, can you also please do the same for freeform_tags?

Reply:
Yes, it is exactly as you said: specifying tags for a particular node pool can override the default ones.
In the latest commit I added the same behavior also for freeform_tags.